app/src/main/java/com/cornellappdev/android/eatery/ui/components/general/FavoriteButton.kt

```diff
@@ -0,0 +1,35 @@
+package com.cornellappdev.android.eatery.ui.components.general
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Icon
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Star
+import androidx.compose.material.icons.outlined.StarOutline
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.unit.dp
+import com.cornellappdev.android.eatery.ui.theme.GrayFive
+import com.cornellappdev.android.eatery.ui.theme.Yellow
+
+@Composable
+fun FavoriteButton(
+    isFavorite: Boolean,
+    onFavoriteClick: (Boolean) -> Unit
+) {
+    Icon(
+        imageVector = if (isFavorite) Icons.Filled.Star else Icons.Outlined.StarOutline,
+        tint = if (isFavorite) Yellow else GrayFive,
+        modifier = Modifier
+            .padding(top = 3.dp)
+            .clip(RoundedCornerShape(9.dp))
+            .clickable(
+                onClick = {
+                    onFavoriteClick(!isFavorite)
+                }
+            ),
+        contentDescription = null
+    )
+}
```

Comment:
Oh weird, surprised we didn't already have this. I'm guessing this wasn't used in other places properly? Like on the eatery card and stuff?

Small nit: Let's take in an optional `modifier: Modifier = Modifier` to make this feel like a legit component (and then replace the `Modifier` in the `Icon` with `modifier`), since this feels pretty generally useful.

Reply:
yeah the code was just duplicated so I abstracted it. I'll add a modifier 👍